crates/network/src/ethereum/mod.rs

```diff
@@ -0,0 +1,36 @@
+use crate::{Network, ReceiptResponse};
+
+mod builder;
+
+mod signer;
+pub use signer::EthereumSigner;
+
+/// Types for a mainnet-like Ethereum network.
+#[derive(Debug, Clone, Copy)]
+pub struct Ethereum {
+    _private: (),
+}
+
+impl Network for Ethereum {
+    type TxEnvelope = alloy_consensus::TxEnvelope;
+
+    type UnsignedTx = alloy_consensus::TypedTransaction;
+
+    type ReceiptEnvelope = alloy_consensus::ReceiptEnvelope;
+
+    type Header = alloy_consensus::Header;
+
+    type TransactionRequest = alloy_rpc_types::transaction::TransactionRequest;
+
+    type TransactionResponse = alloy_rpc_types::Transaction;
+
+    type ReceiptResponse = alloy_rpc_types::TransactionReceipt;
+
+    type HeaderResponse = alloy_rpc_types::Header;
+}
+
+impl ReceiptResponse for alloy_rpc_types::TransactionReceipt {
+    fn contract_address(&self) -> Option<alloy_primitives::Address> {
+        self.contract_address
+    }
+}
```

Comment:
i forget why we cannot put this on the main `TxReceipt` trait

Reply:
Standard ethereum receipts do not have any tx sender field + nonce, so we can't recompute the created address. They also don't include this info, they only have logs/bloom/status/root